src/types/Graph.ts

```diff
@@ -82,3 +82,94 @@ export interface CytoscapeBaseEvent {
 export interface CytoscapeClickEvent extends CytoscapeBaseEvent {}
 export interface CytoscapeMouseInEvent extends CytoscapeBaseEvent {}
 export interface CytoscapeMouseOutEvent extends CytoscapeBaseEvent {}
+
+// Graph Structures
+export interface ProtocolTraffic {
+  protocol: string;
+  rates: any;
+}
+
+export interface GraphNodeData {
+  id: string;
+  parent?: string;
+  nodeType: NodeType;
+  namespace: string;
+  workload?: string;
+  app?: string;
+  version?: string;
+  service?: string;
+  destServices?: any;
+  traffic?: ProtocolTraffic[];
```

Comment:
As far as I know the traffic field is not needed, at least not at the moment.  The approach is that the traffic structure in the JSON is parsed out into the top level fields that you have moved into DecoratedGraphNodeData.

Reply:
Yes, but this structure represents the "undecorated" node data, so I understand that this is not the way that the structure is used on the code because it uses the decorated version.